storm-core/src/jvm/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -121,7 +116,7 @@ public SchedulingResult schedule(TopologyDetails td) {
                     td.getTaskResourceReqList(exec), entry.getKey() });
                     WorkerSlot targetSlot = this.findWorkerForExec(exec, td, schedulerAssignmentMap);
                     if (targetSlot != null) {
-                        RAS_Node targetNode = this.idToNode(targetSlot.getNodeId());
+                        NodeDetails targetNode = this.idToNode(targetSlot.getNodeId());
```

Comment:
It seems there is a chunk of duplicate code here and in the for loop below.  We are adding one line to both, but maybe it would be better to pull them out into a helper method to avoid accidental mistakes in the future?